src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -73,10 +74,82 @@ export const ValidatorName = ({ moniker, isStaking }: ValidatorNameProps) => {
   )
 }
 
-export const StakingOpportunities = ({ assetId, opportunities }: StakingOpportunitiesProps) => {
-  const isLoaded = true
-  const isStaking = opportunities.some(x => x.cryptoAmount)
-  const assetSymbol = useAppSelector(state => selectAssetByCAIP19(state, assetId)).symbol
+export const StakingOpportunities = ({ assetId }: StakingOpportunitiesProps) => {
+  const stakingDataStatus = useAppSelector(selectStakingDataStatus)
+  const isStakingDataLoaded = stakingDataStatus === 'loaded'
+  const validatorStatus = useAppSelector(selectValidatorStatus)
+  const isValidatorDataLoaded = validatorStatus === 'loaded'
+  const isLoaded = isStakingDataLoaded && isValidatorDataLoaded
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+  const dispatch = useAppDispatch()
+
+  const accountSpecifiersForChainId = useAppSelector(state =>
+    selectPubkeyishByChainId(state, asset?.caip2)
+  )
+  const accountSpecifier = accountSpecifiersForChainId?.[0]
+
+  const stakingOpportunities = useAppSelector(state =>
+    selectStakingOpportunityDataByDenom(
+      state,
+      accountSpecifier,
+      SHAPESHIFT_VALIDATOR_ADDRESS,
+      ASSET_ID_TO_DENOM[asset.caip19]
+    )
+  )
+  const shapeshiftValidator = useAppSelector(state =>
+    selectSingleValidator(state, accountSpecifier, SHAPESHIFT_VALIDATOR_ADDRESS)
+  )
+  const stakingOpportunityDefault = [
+    {
+      ...shapeshiftValidator
+    }
+  ]
+  const nonLoadedValidators = useAppSelector(state =>
+    selectNonloadedValidators(state, accountSpecifier)
+  )
+  const isStaking = stakingOpportunities.length !== 0
```

Comment:
[gomesalexandre](https://github.com/gomesalexandre) [7 hours ago](https://github.com/shapeshift/web/pull/1356/files#r840739416)
I wouldn't call these stakingOpportunities if they're supposed to refer to the active set. activeStakingOpportunities might be a better fit

Reply:
agree stakingOpportunities is kindof a weird name for this.  Looking at the selector it feels more related to validators?

Reply:
> agree stakingOpportunities is kindof a weird name for this. Looking at the selector it feels more related to validators?

No, I guess it's related to active staking opportunities, that you can find in the Cosmos Overview modal